File: java/planefinder/src/main/java/com/thehecklers/planefinder/SecurityConfig.java
package com.thehecklers.planefinder;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.web.server.ServerHttpSecurity;
import org.springframework.security.web.server.SecurityWebFilterChain;

@Configuration
public class SecurityConfig {
    @Bean
    public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
        http
                .authorizeExchange()
                .pathMatchers("/aircraft/**").hasAuthority("SCOPE_closedid")
                .pathMatchers("/aircraftadmin/**").hasAuthority("SCOPE_openid")
                .and().oauth2ResourceServer().jwt();

        return http.build();
    }
}

File: chapter10end/kotlin/aircraft-positions/src/main/kotlin/com/thehecklers/aircraftpositions/SecurityConfig.kt
package com.thehecklers.aircraftpositions

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository
import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class SecurityConfig {
    @Bean
    fun client(regRepo: ClientRegistrationRepository, cliRepo: OAuth2AuthorizedClientRepository): WebClient {
        val filter = ServletOAuth2AuthorizedClientExchangeFilterFunction(regRepo, cliRepo)

        filter.setDefaultOAuth2AuthorizedClient(true)

        return WebClient.builder()
            .baseUrl("http://localhost:7634/")
            .apply(filter.oauth2Configuration())
            .build()
    }
}